src/Features/Core/Portable/Common/TaggedText.cs

```diff
@@ -47,14 +80,42 @@ public override string ToString()
     internal static class TaggedTextExtensions
     {
         public static ImmutableArray<TaggedText> ToTaggedText(this IEnumerable<SymbolDisplayPart> displayParts)
+            => displayParts.ToTaggedText(TaggedTextStyle.None);
+
+        public static ImmutableArray<TaggedText> ToTaggedText(this IEnumerable<SymbolDisplayPart> displayParts, TaggedTextStyle style)
         {
             if (displayParts == null)
             {
                 return ImmutableArray<TaggedText>.Empty;
             }
 
-            return displayParts.Select(d =>
-                new TaggedText(SymbolDisplayPartKindTags.GetTag(d.Kind), d.ToString())).ToImmutableArray();
+            return displayParts.SelectAsArray(d =>
+                new TaggedText(
+                    SymbolDisplayPartKindTags.GetTag(d.Kind),
+                    d.ToString(),
+                    style,
+                    GetNavigationTarget(d.Symbol),
+                    GetNavigationHint(d.Symbol)));
+
+            static string GetNavigationTarget(ISymbol symbol)
+            {
+                if (symbol is null)
+                {
+                    return null;
+                }
+
+                return SymbolKey.ToString(symbol);
+            }
+
+            static string GetNavigationHint(ISymbol symbol)
+            {
+                if (symbol is null)
+                {
+                    return null;
+                }
+
+                return symbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat);
```

Comment:
nit: consider simplifying to `symbol?.ToDis...`

Reply:
I prefer the current form since the lack of a symbol is the primary case.